tests/components/configurator/test_init.py

```diff
@@ -1,117 +1,106 @@
-            self.hass, "Test Request", lambda _: None
-        )
-        configurator.request_done(self.hass, request_id)
-        assert 1 == len(self.hass.states.all())
-
-        self.hass.bus.fire(EVENT_TIME_CHANGED)
-        self.hass.block_till_done()
-        assert 0 == len(self.hass.states.all())
-
-    def test_request_done_fail_silently_on_bad_request_id(self):
-        """Test that request_done fails silently with a bad request id."""
-        configurator.request_done(self.hass, 2016)
+        configurator.ATTR_SUBMIT_CAPTION: "config submit caption",
+        configurator.ATTR_FIELDS: [],
+        configurator.ATTR_ENTITY_PICTURE: "config entity picture",
+        configurator.ATTR_CONFIGURE_ID: configurator.async_request_config(
+            hass,
+            name="Test Request",
+            callback=lambda _: None,
+            description="config description",
+            description_image="config image url",
+            submit_caption="config submit caption",
+            fields=None,
+            link_name="link name",
+            link_url="link url",
+            entity_picture="config entity picture",
+        ),
+    }
+
+    states = hass.states.async_all()
+    assert 1 == len(states)
+    state = states[0]
+
+    assert configurator.STATE_CONFIGURE == state.state
+    assert exp_attr == state.attributes
+
+
+async def test_callback_called_on_configure(hass):
+    """Test if our callback gets called when configure service called."""
+    calls = []
+    request_id = configurator.async_request_config(
+        hass, "Test Request", lambda _: calls.append(1)
+    )
+
+    await hass.services.async_call(
+        configurator.DOMAIN,
+        configurator.SERVICE_CONFIGURE,
+        {configurator.ATTR_CONFIGURE_ID: request_id},
+    )
+
+    await hass.async_block_till_done()
+    assert 1 == len(calls), "Callback not called"
+
+
+async def test_state_change_on_notify_errors(hass):
+    """Test state change on notify errors."""
+    request_id = configurator.async_request_config(hass, "Test Request", lambda _: None)
+    error = "Oh no bad bad bad"
+    configurator.async_notify_errors(hass, request_id, error)
+
+    states = hass.states.async_all()
+    assert 1 == len(states)
+    state = states[0]
+    assert error == state.attributes.get(configurator.ATTR_ERRORS)
+
+
+async def test_notify_errors_fail_silently_on_bad_request_id(hass):
+    """Test if notify errors fails silently with a bad request id."""
+    configurator.async_notify_errors(hass, 2015, "Try this error")
+
+
+async def test_request_done_works(hass):
+    """Test if calling request done works."""
+    request_id = configurator.async_request_config(hass, "Test Request", lambda _: None)
+    configurator.async_request_done(hass, request_id)
+    assert 1 == len(hass.states.async_all())
+
+    hass.bus.async_fire(EVENT_TIME_CHANGED)
+    await hass.async_block_till_done()
+    assert 0 == len(hass.states.async_all())
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)